### PR TITLE
Update and move some Query filters

### PR DESCRIPTION
**Pull request by ntsekouras**
## Description
Based on many iterations for Query block design here: https://github.com/WordPress/gutenberg/issues/25198 and slack channels etc, it seems that filtering will end up in the inspector controls. Although the final design has not yet been found and approved, I think it makes sense to move some of the options we already have in the inspector control to make the Query toolbar less crowded.

This PR also makes a couple of changes.

1. Change the default `template` used in Query block from [PostTile, PostContent] to [PostTitle, PostDate, PostExcerpt] that seems to make more sense IMO.
2. Fix a small bug for waiting the results of `tags and categories` to show the input control.
3. Change the inputs `posts per page` and `offset` in `NumberControls`.
4. Gets the default `posts_per_page` number from options.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/query/editor.scss [hunk 1]
```diff
@@ -1,3 +1,13 @@
 .editor-styles-wrapper .wp-block.wp-block-query {
 	max-width: 100%;
 }
+
+.block-library-query-toolbar__popover {
+	.components-number-control {
```
Comment (mtias) [inline packages/block-library/src/query/editor.scss:6]: Are these styles that should be working by default?

Comment (ntsekouras) [reply packages/block-library/src/query/editor.scss:6]: Probably yes. I've ping @ItsJonQ about this and will have more feedback if they should come from NumberControl.

**Comment by ItsJonQ** [comment]
@ntsekouras Haiii! Thanks for looping me into this one.

I can see where you had some trouble with the CSS alignment/widths for these controls x labels.
Inputs and labels have always been a bit random 🙈 . (It's something that I'm hoping will be remedied with the [G2 component implementation](https://github.com/ItsJonQ/g2/blob/master/packages/components/types/components/FormGroup.d.ts#L25))

**Short note:**

Screen Shot 2020-09-28 at 11 44 37 AM

I've made some updates to the base `InputControl` to accept an `labelPosition` value of `edge`, snapping the label and control to the "edges". A new `__unstableInputWidth` value has been added to allow users to fine-tune widths for the input.

Lastly, spacing for controls have always been determined with `<BaseControl />`. I've made the update to wrap the items with that in your updated Toolbar UI.

Ideally, you should not have to write custom `.scss` styles for these things.
Styling (e.g. alignment, width, etc...) should be something that can be adjusted via component props.

**Longer notes:**

The coupling of labels and controls within Gutenberg's input-based components are too tightly coupled. This leads to either awkward or limited alignment and width styling (both of which you experienced).

Ideally, the widths and alignment for these controls should be standardized, following some sort of grid-based layout (predetermined by the UI system). For advance/custom use cases, the individual "parts" (e.g. `Label` and `InputControl`) should be made available as separate pieces, allowing for the user to compose + refine them as needed.

**Review by aristath — APPROVED** [review]
Tested code and confirmed it works.
I did have trouble with an [FSE theme](https://github.com/aristath/q/blob/1c6322549337e79abc525e5324915ae10ab521be/block-templates/index.html) and got the `This block has encountered an error and cannot be previewed.` error on the site-editor screen, but I suppose that is OK?
Since FSE is experimental and this block is also experimental and only available when the FSE experiment is enabled, we don't really have to worry about backwards-compatibility.

Code looks good to me :+1:

**Comment by ntsekouras** [comment]
Hey @aristath - thanks for reviewing.

>I did have trouble with an FSE theme and got the This block has encountered an error and cannot be previewed. error on the site-editor screen, but I suppose that is OK?

Do you mean with a previous existed query block? I'd like to have a look at it.

>Since FSE is experimental and this block is also experimental and only available when the FSE experiment is enabled, we don't really have to worry about backwards-compatibility.

Yes, I think we are okay, since there is going to be a lot of changes soon in Query block.

**Comment by aristath** [comment]
> Do you mean with a previous existed query block? I'd like to have a look at it.

The issue was in the query block I had in [my theme's `index.html` file (link)](https://github.com/aristath/q/blob/1c6322549337e79abc525e5324915ae10ab521be/block-templates/index.html), you can just install the theme if you want and test it there, or copy-paste the contents of the query in a new post on any theme:
```html
<!-- wp:query {"queryId":1,"query":{"perPage":10,"offset":0,"orderBy":"date"}} -->
<!-- wp:query-loop -->
<!-- wp:post-title {"isLink":true,"linkTarget":"_self"} /-->

<!-- wp:post-content /-->
<!-- /wp:query-loop -->

<!-- wp:query-pagination /-->
<!-- /wp:query -->
```

**Pull request merged**